### PR TITLE
[Hub Generated] Review request for Microsoft.ApiManagement to add version preview/2019-12-01-preview

### PR DESCRIPTION
**Pull request by solankisamir**
If you are a MSFT employee you can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/solankisamir/azure-rest-api-specs/solanki%2Fapim-2019-12-01-preview...master/).

### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by solankisamir** [comment]
This is PR related to https://github.com/Azure/azure-rest-api-specs/pull/8046. We couldn't the apim-2019-12-01-preview branch as it was read-only. Hence created a clone here and working on edits here.

**Comment by solankisamir** [comment]
@NullMDR I am trying to fix the prettier issues using https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/ci-fix.md, but I am getting this error 

> D:\github\azure-rest-api-specs>npm install prettier-fix
npm ERR! code E404
npm ERR! 404 Not Found - GET https://registry.npmjs.org/prettier-fix - Not found
npm ERR! 404
npm ERR! 404  'prettier-fix@latest' is not in the npm registry.
npm ERR! 404 You should bug the author to publish it (or use the name yourself!)
npm ERR! 404
npm ERR! 404 Note that you can also install from a
npm ERR! 404 tarball, folder, http url, or git url.

> npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\sasolank\AppData\Roaming\npm-cache\_logs\2020-01-05T19_20_03_381Z-debug.log

**Comment by NullMDR** [comment]
@solankisamir You need 

npm install; npm run prettier -- --write "specification/apimanagement/**/*.json"

**Comment by solankisamir** [comment]
@NullMDR ran the command and hit the issue below. I dont have the `azure-rest-api-specs-test` repos locally. Do I need to get that?
[detailed_log](https://gist.github.com/solankisamir/5266d504d167221744e3f8250ce4a1f5)

> PS D:\github\azure-rest-api-specs> npm install; npm run prettier -- --write "specification/apimanagement/**/*.json"

> azure-rest-api-specs-tests@0.1.0 postinstall D:\github\azure-rest-api-specs
> scripts/switch-to-preproduction.sh
npm ERR! errno 1
npm ERR! azure-rest-api-specs-tests@0.1.0 postinstall: `scripts/switch-to-preproduction.sh`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the azure-rest-api-specs-tests@0.1.0 postinstall script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

> npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\sasolank\AppData\Roaming\npm-cache\_logs\2020-01-06T16_40_16_473Z-debug.log
npm ERR! missing script: prettier

> npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\sasolank\AppData\Roaming\npm-cache\_logs\2020-01-06T16_40_18_669Z-debug.log

**Review by ryansbenson — COMMENTED** [review]
One minor change

File: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2019-12-01-preview/definitions.json
Comment (vfedonkin) [reply specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2019-12-01-preview/definitions.json]: It's on the top level, not inside "properties".

**Comment by NullMDR** [comment]
@solankisamir Please merge from latest master to pick up latest scripts and try again

File: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2019-12-01-preview/definitions.json
Comment (solankisamir) [reply specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2019-12-01-preview/definitions.json]: @ryansbenson please see @vfedonkin comment. This is a contract associated with a proxy resource and we are just adding `id` to fix the ARM warnings generated as part of live traffic.

Comment (vfedonkin) [reply specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2019-12-01-preview/definitions.json]: @ryansbenson, @solankisamir , I think we will need to modify all entities to have our fields inside "properties", but it should be done in our next version, since we already behind the schedule for the version we trying to release.

**Review by mmyyrroonn — APPROVED** [review]
LGTM. wait for ARM's signoff

File: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2019-12-01-preview/definitions.json
Comment (solankisamir) [reply specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2019-12-01-preview/definitions.json]: @ryansbenson a gentle ping on this!

File: specification/apimanagement/resource-manager/readme.md
Comment (solankisamir) [reply specification/apimanagement/resource-manager/readme.md]: fixed

File: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2019-12-01-preview/definitions.json
Comment (solankisamir) [reply specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2019-12-01-preview/definitions.json]: @ryansbenson we fixed this in the spec to unblock it.

**Review by vfedonkin — APPROVED** [review]

**Comment by MyronFanQiu** [comment]
@ryansbenson sorry! I made a mistake to merge it. Could you please review this PR, if still need some change we can open another or revert this one. @solankisamir Sorry. Please hold on the following process since ARM didn't sign off yet. Thanks.

**Comment by NullMDR** [comment]
@MyronFanQiu You can create a PR to revert this one

**Comment by sanjaiganesh** [comment]
>           }


For all async operations, add x-ms-long-running-operation:true and

, add "x-ms-long-running-operation-options" extension with "final-state-via" property.
``` json
"x-ms-long-running-operation-options": {
  "final-state-via": "location"
}
```

or

``` json
"x-ms-long-running-operation-options": {
  "final-state-via": "azure-async-operation"
}
```

---
Refers to: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2019-12-01-preview/apimapis.json:2361 in 13fb704. [](commit_id = 13fb70433aee59b3ec37ef3f54a162530abca46a, deletion_comment = False)

File: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2019-12-01-preview/apimnamedvalues.json
Comment (sanjaiganesh) [inline specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2019-12-01-preview/apimnamedvalues.json]: For all async operations, add x-ms-long-running-operation:true and

, add "x-ms-long-running-operation-options" extension with "final-state-via" property.
``` json
"x-ms-long-running-operation-options": {
  "final-state-via": "location"
}
```

or

``` json
"x-ms-long-running-operation-options": {
  "final-state-via": "azure-async-operation"
}
```

File: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2019-12-01-preview/apimnamedvalues.json
Comment (sanjaiganesh) [inline specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2019-12-01-preview/apimnamedvalues.json]: missing ListBySubscrition and ListByResourceGroup calls

File: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2019-12-01-preview/apimnamedvalues.json
Comment (sanjaiganesh) [inline specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2019-12-01-preview/apimnamedvalues.json]: add sync response 200 as well.. as service could respond synchronously if no changes.

File: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2019-12-01-preview/apimnamedvalues.json
Comment (sanjaiganesh) [inline specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2019-12-01-preview/apimnamedvalues.json]: check other comments on LRO tracking extension annotation

**Comment by sanjaiganesh** [comment]
looks fine.. added few comments. please address them in separate PR. added armsignoff.

**Comment by solankisamir** [comment]
> looks fine.. added few comments. please address them in separate PR. added armsignoff.

@sanjaiganesh Thanks for the review. @vfedonkin can you create another PR with the changes requested.

File: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2019-12-01-preview/apimnamedvalues.json
Comment (vfedonkin) [reply specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2019-12-01-preview/apimnamedvalues.json]: @sanjaiganesh , should we remove this header description and use this instead?:

"x-ms-long-running-operation-options": {
  "final-state-via": "location"
}

File: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2019-12-01-preview/apimnamedvalues.json
Comment (vfedonkin) [reply specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2019-12-01-preview/apimnamedvalues.json]: We use 204 for "update was successful" since no any data returned

File: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2019-12-01-preview/apimnamedvalues.json
Comment (vfedonkin) [reply specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2019-12-01-preview/apimnamedvalues.json]: We don't have such methods, "ListByService" only

**Pull request merged**
